### PR TITLE
Add foldMapM to Free

### PR DESCRIPTION
**Pull request by pakoito**
https://www.youtube.com/watch?v=-QRZpligDOs

### CODE REVIEW

File: kategory/src/main/kotlin/kategory/free/Free.kt [hunk 1]
```diff
@@ -6,6 +6,9 @@ typealias FreeF<S> = HK<Free.F, S>
 fun <S, A> FreeKind<S, A>.ev(): Free<S, A> =
         this as Free<S, A>
 
+fun <M, S, A> FreeKind<S, A>.foldMapM(f: FunctionK<S, M>, MM: Monad<M>): HK<M, A> =
+        (this as Free<S, A>).foldMap(f, MM)
```
Comment (raulraja) [inline kategory/src/main/kotlin/kategory/free/Free.kt]: Should this be `this.ev().foldMap(f, MM)` ? 
Also foldMapM has other semantics in Scala which is foldmapping along with the monad. I believe this should be just foldMap if it does not colides with foldMap in Free

Comment (pakoito) [inline kategory/src/main/kotlin/kategory/free/Free.kt]: It does collide.

The ev() jump is just a perf hit for the sake of it, I've been inlining it for now.

Comment (pakoito) [inline kategory/src/main/kotlin/kategory/free/Free.kt]: FoldMapK

File: kategory/src/main/kotlin/kategory/free/Free.kt [hunk 2]
```diff
@@ -6,6 +6,9 @@ typealias FreeF<S> = HK<Free.F, S>
 fun <S, A> FreeKind<S, A>.ev(): Free<S, A> =
         this as Free<S, A>
 
+fun <M, S, A> FreeKind<S, A>.foldMapK(f: FunctionK<S, M>, MM: Monad<M>): HK<M, A> =
+        (this as Free<S, A>).foldMap(f, MM)
```
Comment (ffgiraldez) [inline kategory/src/main/kotlin/kategory/free/Free.kt]: why not ev().foldMap?

Comment (pakoito) [inline kategory/src/main/kotlin/kategory/free/Free.kt]: https://github.com/kategory/kategory/pull/157#discussion_r129730814

Comment (pakoito) [inline kategory/src/main/kotlin/kategory/free/Free.kt]: Microoptimization

**Pull request merged**